constraint/pkg/client/e2e_test.go

```diff
@@ -834,24 +835,103 @@ func TestE2E_Tracing_Unmatched(t *testing.T) {
 	}
 }
 
-// TestE2E_Review_RegoEvaluationMeta tests that we can get stats out of evaluated constraints.
-func TestE2E_Review_RegoEvaluationMeta(t *testing.T) {
+func TestE2E_DriverCfg(t *testing.T) {
+	tests := []struct {
+		name         string
+		statsEnabled bool
+	}{
+		{
+			name:         "disabled",
+			statsEnabled: false,
+		},
+		{
+			name:         "enabled",
+			statsEnabled: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			c := clienttest.New(t)
+
+			_, err := c.AddTemplate(ctx, clienttest.TemplateDeny())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			_, err = c.AddConstraint(ctx, cts.MakeConstraint(t, clienttest.KindDeny, "foo"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			obj := handlertest.Review{Object: handlertest.Object{Name: "bar"}}
+
+			rsps, err := c.Review(ctx, obj, drivers.Stats(tt.statsEnabled))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			stats := rsps.StatsEntries
+			if stats == nil && tt.statsEnabled {
+				t.Fatal("got nil stats but stats enabled for Review")
+			} else if len(stats) != 0 && !tt.statsEnabled {
+				t.Fatal("got stats but stats disabled")
+			}
+
+			_, err = c.AddData(ctx, &obj.Object)
+			if err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+// TestE2E_Review_StatsEntries tests that we can get stats out of evaluated constraints.
+// In particular, this test makes sure we have stats for both violating constraint kind, name pairs
+// and non violating ones.
+func TestE2E_Review_StatsEntries(t *testing.T) {
 	ctx := context.Background()
-	c := clienttest.New(t)
-	ct := clienttest.TemplateCheckData()
-	_, err := c.AddTemplate(ctx, ct)
+	d, err := rego.New(rego.GatherStats())
```

Comment:
We should move away from using Rego to using the fake driver in tests. This will reduce randomness and allow us to do things like inject specific stats responses, error responses, etc.

Reply:
I'm with you with using the `fake` driver for more int tests. In this case, what would be the best way to add in an automated int test for the `Rego` driver, in particular? 

I want to make sure that we get stats out of that one and no future changes to its internal cause a regression to this behavior.

Reply:
Can we write those tests next to the Rego driver, calling it directly?

Reply:
For using the `fake` driver more in the e2e tests I've added: https://github.com/open-policy-agent/frameworks/issues/317 . In the case of this PR, I moved the `rego` tests in the driver package and kept the `DriverCfg` test to make sure that stats are returned when turned on.